include/misc/poset.h

```diff
@@ -25,6 +25,36 @@ class Poset {
     unsigned int n;  // elements will be from 0 to n-1
     RV order_relations;     // pairs of form a <= b
 
+    static void sorted_list(std::vector<unsigned int> &res, const unsigned int &n, const RV &relations)
```

Comment:
I think it is more readable if you reorder parameters such that input parameters are first and output second, i.e., `const unsigned int & n, const RV &relations, std::vector<unsigned int> &res`